src/Marten/Schema/JsonLocatorField.cs

```diff
@@ -14,38 +15,55 @@ public static JsonLocatorField For<T>(EnumStorage enumStyle, Casing casing, Expr
         {
             var property = ReflectionHelper.GetProperty(expression);
 
-            return new JsonLocatorField("d.data", new StoreOptions(), enumStyle, casing, property);
+            return new JsonLocatorField("d.data", null, enumStyle, casing, property);
         }
 
         private readonly Func<Expression, object> _parseObject = expression => expression.Value();
 
-        public JsonLocatorField(string dataLocator, StoreOptions options, EnumStorage enumStyle, Casing casing, MemberInfo member) : base(enumStyle, member)
+        public JsonLocatorField(string dataLocator, string databaseSchemaName, EnumStorage enumStyle, Casing casing, MemberInfo member, string pgType = null) :
+            this(dataLocator, databaseSchemaName, enumStyle, casing, new[] { member }, pgType)
         {
-            var memberType = member.GetMemberType();
-            var memberName = member.Name.FormatCase(casing);
+        }
+
+        public JsonLocatorField(string dataLocator, string databaseSchemaName, EnumStorage enumStyle, Casing casing, MemberInfo[] members, string pgType = null) : base(enumStyle, members)
+        {
+            var locator = new StringBuilder(dataLocator);
```

Comment:
**SUGGESTION:** This location building logic would be nice to move to some separate method/class.

Reply:
Moved out into refactored `CommandBuilder`